generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/ComponentDestinationType.java

```diff
@@ -40,15 +40,39 @@ public ComponentDestinationType() {
      *                                  Component}
      */
     public Class<? extends Destination> inputTypeFor(final String component) {
+        if (component.contains(COMMAND_API)) {
+            return components.get(COMMAND_API);
+        }
+
+        if (component.contains(COMMAND_CONTROLLER)) {
+            return components.get(COMMAND_CONTROLLER);
+        }
+
+        if (component.contains(COMMAND_HANDLER)) {
+            return components.get(COMMAND_HANDLER);
+        }
+
+        if (component.contains(EVENT_PROCESSOR)) {
+            return components.get(EVENT_PROCESSOR);
+        }
+
+        if (component.contains(EVENT_LISTENER)) {
+            return components.get(EVENT_LISTENER);
+        }
+
         if (components.containsKey(component)) {
             return components.get(component);
         }
 
         throw new IllegalArgumentException("No input destination type defined for service component of type " + component);
     }
 
-
     public boolean isSupported(final String component) {
-        return components.containsKey(component);
+        return component.contains(COMMAND_API)
```

Comment:
Isn''t this just always going to return true now?

Reply:
not quite...some components are not on the list of supported components
        COMMAND_API
        COMMAND_CONTROLLER
        COMMAND_HANDLER
        EVENT_PROCESSOR
        EVENT_LISTENER
Are the only components supported plus custom versions of theses.